[jsk_perception] bof_feature_extractor.py with old pickle does not work

If you install sklearn with pip, it raise error with old pickled data. (because of changed api)
```
[ERROR] [WallTime: 1463393760.278256] bad callback: <bound method Subscriber.callback of <message_filters.Subscriber object at 0x7fe54077a1d0>>
Traceback (most recent call last):
  File "/opt/ros/indigo/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
    cb(msg)
  File "/opt/ros/indigo/lib/python2.7/dist-packages/message_filters/__init__.py", line 74, in callback
    self.signalMessage(msg)
  File "/opt/ros/indigo/lib/python2.7/dist-packages/message_filters/__init__.py", line 56, in signalMessage
    cb(*(msg + args))
  File "/opt/ros/indigo/lib/python2.7/dist-packages/message_filters/__init__.py", line 199, in add
    self.signalMessage(*msgs)
  File "/opt/ros/indigo/lib/python2.7/dist-packages/message_filters/__init__.py", line 56, in signalMessage
    cb(*(msg + args))
  File "/home/baxter/ros/indigo/src/jsk-ros-pkg/jsk_recognition/jsk_perception/node_scripts/bof_histogram_extractor.py", line 75, in _apply
    X = self.bof.transform(X)
  File "/home/baxter/ros/indigo/src/jsk-ros-pkg/jsk_recognition/jsk_recognition_utils/python/jsk_recognition_utils/feature.py", line 24, in transform
    return np.vstack([self.make_hist(xi.reshape((-1, 128))) for xi in X])
  File "/home/baxter/ros/indigo/src/jsk-ros-pkg/jsk_recognition/jsk_recognition_utils/python/jsk_recognition_utils/feature.py", line 31, in make_hist
    indices = nn.kneighbors(descriptors, return_distance=False)
  File "/home/baxter/.local/lib/python2.7/site-packages/sklearn/neighbors/base.py", line 366, in kneighbors
    n_jobs = _get_n_jobs(self.n_jobs)
AttributeError: 'NearestNeighbors' object has no attribute 'n_jobs'
```

--- a/jsk_perception/node_scripts/bof_histogram_extractor.py
+++ b/jsk_perception/node_scripts/bof_histogram_extractor.py
@@ -6,6 +6,8 @@ Extract BoF histogram in realtime.
 
 import gzip
 import cPickle as pickle
+from distutils.version import StrictVersion
+from pkg_resources import get_distribution
 
 import numpy as np
 from sklearn.preprocessing import normalize
@@ -33,6 +35,13 @@ class BoFHistogramExtractor(ConnectionBasedTransport):
             quit()
         with gzip.open(bof_data, 'rb') as f:
             self.bof = pickle.load(f)
+        if (StrictVersion(get_distribution('scikit-learn').version) >=
+                StrictVersion('0.17.0')):
+            if 'n_jobs' not in bof.nn.__dict__:
+                # In scikit-learn>=0.17.0,
+                # sklearn.neighbors.NearestNeighbors needs 'n_jobs' attribute.
+                # https://github.com/jsk-ros-pkg/jsk_recognition/issues/1669
+                bof.nn.n_jobs = 1
 
         self._pub = self.advertise('~output', VectorArray, queue_size=1)
         jsk_loginfo('Initialized BoF histogram extractor')

